### PR TITLE
BUG: fix parsing by filtering out junk lines

### PR DESCRIPTION
**Pull request by bsipocz**
This closes https://github.com/astropy/astroquery/issues/2717

I'm on the fence whether this needs a changelog or not, I suppose it does as lines from the docs stopped working due to these junk lines, e.g. this one 
```
>>> response['FREQ'].quantity
```

### CODE REVIEW

File: docs/jplspec/jplspec.rst [hunk 1]
```diff
@@ -235,7 +234,7 @@ response length:
 .. doctest-remote-data::
 
    >>> print(result.meta['comments'])
-   ['', '', '', '', '', 'form is currently limited to 2000 lines. Please limit your search.']
+   ['', '', '', '', '', '', '', '', '', '', '', 'form is currently limilted to 2000 lines. Please limit your search.']
```
Comment (keflavich) [inline docs/jplspec/jplspec.rst:237]:
```suggestion
   ['', '', '', '', '', '', '', '', '', '', '', 'form is currently limited to 2000 lines. Please limit your search.']
```

**Review by keflavich — APPROVED** [review]
except for minor typo, looks great

**Comment by mkelley** [comment]
This looks good to me, except that the LGINT units comparison failed for me locally:
```
Package versions: 
Numpy: 1.21.4
Scipy: 1.7.1
Matplotlib: 3.5.2
h5py: 3.4.0
Pandas: 1.3.3
Astropy: 5.0.1
regions: not available
pyVO: not available
mocpy: not available
astropy-healpix: 0.6
vamdclib: not available
astropy-helpers: 4.0.1

Using Astropy options: remote_data: any.

rootdir: /home/msk/Projects/astroquery, configfile: setup.cfg
plugins: cov-2.12.1, typeguard-2.13.3, xdist-2.5.0, arraydiff-0.3, requests-mock-1.9.3, remotedata-0.3.2, hypothesis-6.21.6, anyio-3.6.1, forked-1.4.0, mock-3.6.1, openfiles-0.5.0, filter-subpackage-0.1.1, astropy-header-0.1.2, asdf-2.8.1, doctestplus-0.11.0
collected 1 item                                                                                                                                                                         

docs/jplspec/jplspec.rst F                                                                                                                                                         [100%]

======================================================================================== FAILURES ========================================================================================
_________________________________________________________________________________ [doctest] jplspec.rst __________________________________________________________________________________
027 .. doctest-remote-data::
028 
029    >>> from astroquery.jplspec import JPLSpec
030    >>> import astropy.units as u
031    >>> response = JPLSpec.query_lines(min_frequency=100 * u.GHz,
032    ...                                max_frequency=1000 * u.GHz,
033    ...                                min_strength=-500,
034    ...                                molecule="28001 CO",
035    ...                                get_query_payload=False)
036    >>> print(response)
Differences (unified diff with -expected +actual):
    @@ -1,4 +1,4 @@
    -    FREQ     ERR    LGINT   DR   ELO   GUP  TAG   QNFMT QN' QN"
    -    MHz      MHz   nm2 MHz      1 / cm
    +    FREQ     ERR    LGINT   DR   ELO    GUP  TAG   QNFMT QN' QN"
    +    MHz      MHz   MHz nm2      1 / cm                          
     ----------- ------ ------- --- -------- --- ------ ----- --- ---
     115271.2018 0.0005 -5.0105   2      0.0   3 -28001   101   1   0
```

**Comment by keflavich** [comment]
I think the `lgint` comparison is related to astropy version - possibly this PR:
https://github.com/astropy/astropy/issues?q=order++units

**Review by mkelley — APPROVED** [review]

File: docs/jplspec/jplspec.rst (hunk 1 above)
Comment (bsipocz) [reply docs/jplspec/jplspec.rst:237]: Don't fix the too, this is what comes from the server 😅

Comment (keflavich) [reply docs/jplspec/jplspec.rst:237]: OH! Wow.  🤦

**Comment by bsipocz** [comment]
Oh, great. Then I need to double check whether I use the same version as ci.

**Comment by bsipocz** [comment]
OK, so this is the PR that brake stuff, and the suggested workaround for a global pin in setup.cfg won't do it for us: 
https://github.com/astropy/astropy/pull/14439

We do run remote-data tests for the latest astropy release (5.2.x currently), and with the dev version, so I would vote for making these changes for the dev test and live with the breaks for the windows online tests until 5.3 comes out rather than adding hacks.

This change affects the `jplspec` module, and `cdms`, and only the narrative documentation. Fix for the new formatting is in this PR and in https://github.com/astropy/astroquery/pull/2721.

**Comment by bsipocz** [comment]
Side note: I'm very much looking forward to use https://github.com/astropy/pytest-doctestplus/pull/197, so our rst doctesting won't be bogged down at a doctest failure for a print command.

**Pull request merged**